encrypted-media/scripts/playback-temporary-setMediaKeys.js

```diff
@@ -64,20 +83,24 @@ function runTest(config) {
         }).then(function(mediaKeys) {
             _mediaKeys = mediaKeys;
 
-            _video.setMediaKeys(_mediaKeys);
-
             _mediaKeySession = _mediaKeys.createSession( 'temporary' );
 
             waitForEventAndRunStep('encrypted', _video, onEncrypted, test);
             waitForEventAndRunStep('playing', _video, onPlaying, test);
-        }).then(function() {
+        }).then(test.step_func(function() {
+            if ( config.testcase === SETMEDIAKEYS_FIRST ) {
```

Comment:
"FIRST" isn't really first. I would assume that means before a session is created. IOW, at line 85. (That would mean moving everything else in that block into this .then() block.

Also, FIRST is ambiguous. Maybe s/FIRST/IMMEDIATELY/.